youtube_dl/extractor/masterclass.py

```diff
@@ -0,0 +1,77 @@
+# coding: utf-8
+from __future__ import unicode_literals
+import re
+
+from .brightcove import BrightcoveNewIE
+from .common import InfoExtractor
+
+
+class MasterClassIE(InfoExtractor):
+    IE_NAME = 'masterclass'
+    _VALID_URL = r'https:\/\/www.masterclass.com[\/]classes[\/](.+?)(?=[\/]|$)(?:[\/]chapters)?(?:[\/])?(.+?(?=[?]|$))?'
```

Comment:
This is total mess. You don't need to used braces for single character, you don't need to escape forward slash, you don't need positive lookahead here, you don't need optional groups.